src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -65,6 +65,11 @@ public IReadOnlyList<PackageSpec> Projects
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating if this instance can be treated as read-only and no other threads will update it.
+        /// </summary>
+        public bool IsReadOnly { get; set; }
```

Comment:
Should we actually make this something that can be set in the constructor only? 

Let's not add unnecessary headers if we don't have to.